bindings/nodejs/lib/spi.js

```diff
@@ -64,10 +64,12 @@ _.extend( SPIBus.prototype, {
                function( txData, rxData, count ) {
                    if ( rxData !== null ) {
                        fulfill( rxData );
+                   } else {
+                       reject( new Error( "SPI transmission failed" ) );
                    }
            });
 
-           if ( ( typeof returnStatus === 'undefined' ) || !returnStatus ) {
+           if ( ( typeof returnStatus === 'undefined' ) || returnStatus < 0) {
```

Comment:
You don't need to worry about a ```returnStatus``` of ```undefined```, because that can only happen if an exception is thrown, in which case JS execution will jump to the ```reject()``` handler and ```returnStatus``` will not be assigned at all, and this code will not be reached.